packages/core/src/overlay/OverlayHeader.tsx

```diff
@@ -0,0 +1,84 @@
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import clsx from "clsx";
+import {
+  type ComponentPropsWithoutRef,
+  type ReactNode,
+  forwardRef,
+} from "react";
+import { H3, Text } from "../text";
+import { makePrefixer } from "../utils";
+import overlayHeaderCss from "./OverlayHeader.css";
+
+const withBaseName = makePrefixer("saltOverlayHeader");
+
+export interface OverlayPanelContentProps
+  extends ComponentPropsWithoutRef<"div"> {
+  /**
+   * Description text is displayed just below the header
+   **/
+  description?: string;
```

Comment:
There was a question on the previous PR about why `description|header|preheader` were limited to strings. (same applies to DialogHeader)

IMO string is the right choice here. The design is very deliberate about only allowing text content in these sections, and I don't think there's a scenario where we would want users to be able to inject any component they like here.

@joshwooding @origami-z @navkaur76 thoughts?

Reply:
In most cases, pure string would be provided anyway. However, in case teams want to put additional layout, icon or other relevant stuff, we could try to accommodate? A few other components does this

Reply:
I would rather use ReactNode or something similar. It's not worth being restrictive